pkg/sql/logictest/testdata/logic_test/alter_table

```diff
@@ -2014,3 +2014,34 @@ ALTER TABLE t ALTER COLUMN b SET DEFAULT 10
 
 statement error identity column type must be INT, INT2, INT4 or INT8
 ALTER TABLE t ALTER COLUMN b TYPE numeric(10,2)
+
+# Test we can assign a PRIMARY KEY overriding the existing rowid PRIMARY KEY.
+statement ok
+DROP TABLE IF EXISTS t;
+CREATE TABLE t (id INT NOT NULL)
+
+statement ok
+ALTER TABLE t ADD CONSTRAINT t_pkey PRIMARY KEY (id)
+
+query TT
+SHOW CREATE TABLE t
+----
+t  CREATE TABLE public.t (
+   id INT8 NOT NULL,
+   rowid INT8 NOT VISIBLE NOT NULL DEFAULT unique_rowid(),
+   CONSTRAINT t_pkey PRIMARY KEY (id ASC),
+   FAMILY "primary" (id, rowid)
+)
+
+# Table has a PRIMARY KEY named index; check it errors when assinging PRIMARY KEY to id.
+statement ok
+DROP TABLE IF EXISTS t;
+CREATE TABLE t (
+  id INT NOT NULL,
+  rowid INT8 NOT VISIBLE NOT NULL DEFAULT unique_rowid(),
+  CONSTRAINT t_pk PRIMARY KEY (rowid),
+  CONSTRAINT t_pkey UNIQUE (id)
+)
+
+statement error constraint with name "t_pkey" already exists
+ALTER TABLE t ADD CONSTRAINT t_pkey PRIMARY KEY (id)
```

Comment:
Our abstractions around `IsPrimaryIndexDefaultRowID` are pretty crappy. I feel like the decision to always show these automatically created elements but then to detect whether they are the automatically created elements in an ad-hoc way is sort of annoying. Like what allows us to trigger this special behavior of deleting `t_pk` and adding the new primary column is just that we have a hidden column that's a the primary key column and there's only one. Like what the heck is that about? Imagine I did this:

```sql
CREATE TABLE t (
  id INT NOT NULL,
  my_beloved_column INT8 NOT VISIBLE NOT NULL,
  CONSTRAINT t_pk_that_provides_nice_uniqueness PRIMARY KEY (my_beloved_column),
)
```
```
> alter table t add primary key (id);
NOTICE: primary key changes are finalized asynchronously; further schema changes on this table may be restricted until the job completes
ALTER TABLE


Time: 136ms total (execution 136ms / network 0ms)

demo@127.0.0.1:63578/defaultdb> show create table t;
  table_name |                 create_statement
-------------+---------------------------------------------------
  t          | CREATE TABLE public.t (
             |     id INT8 NOT NULL,
             |     my_beloved_column INT8 NOT VISIBLE NOT NULL,
             |     CONSTRAINT "primary" PRIMARY KEY (id ASC),
             |     FAMILY "primary" (id, my_beloved_column)
             | )
```

My feeling is that this code is bad mostly because `IsPrimaryIndexDefaultRowID` is bad.

Reply:
Yeah I mean unless we store a bool for detecting this or add a check that rowid is a prefix of the col name not sure we can do better. Anything you want me to action here?

Reply:
We could look at the type, the name, and the name of the primary key constraint. Thoughts?

Reply:
My only hang up with PK constraint name is renaming the table. Rest I think is agreeable

Reply:
> My only hang up with PK constraint name is renaming the table. Rest I think is agreeable

That's legit. Also the fact that we renamed what we call it now. How about also the existence of a default value for the thing. It'd be weird for it to not have a default value.

Reply:
all done